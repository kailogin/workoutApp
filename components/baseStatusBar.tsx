import React from "react";
import { StatusBar, View } from "react-native";
import { Colors } from "../utils/theme";

export const BaseStatusBar = () => {
  // --- RENDER ---

  return (
    <View>
      <StatusBar
        animated={true}
        backgroundColor={Colors.BLACK}
        barStyle="light-content"
        hidden={false}
      />
    </View>
  );
};
