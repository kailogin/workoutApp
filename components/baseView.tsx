import React, { ReactNode } from "react";
import {
  ScrollView,
  View,
  StyleSheet,
  SafeAreaView,
  ViewStyle,
} from "react-native";

import { Colors } from "../utils/colors";
import { BaseStatusBar } from "./baseStatusBar";

interface BaseViewProps {
  baseViewStyle?: ViewStyle;
  children: ReactNode;
}

export const BaseView = ({ baseViewStyle, children }: BaseViewProps) => {
  return (
    <SafeAreaView style={styles.safe_area_container}>
      <BaseStatusBar />

      <View style={[styles.container, baseViewStyle]}>
        <ScrollView>{children}</ScrollView>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safe_area_container: {
    flex: 1,
  },
  container: {
    backgroundColor: Colors.BLACK,
    flex: 1,
    padding: 40,
  },
});
