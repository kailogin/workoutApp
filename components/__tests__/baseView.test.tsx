import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { BaseView } from "../baseView";
import { BaseStatusBar } from "../baseStatusBar";

const children = <BaseStatusBar />;

test(`BaseView snapshot`, () => {
  const snap = renderer.create(<BaseView>{children}</BaseView>).toJSON();

  expect(snap).toMatchSnapshot();
});
