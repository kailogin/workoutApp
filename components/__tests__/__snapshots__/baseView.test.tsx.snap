// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseView snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flex": 1,
          "padding": 40,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
